business/istio_validations.go

```diff
@@ -480,6 +473,62 @@ func (in *IstioValidationsService) fetchDetails(rValue *kubernetes.IstioDetails,
 	}
 }
 
+func (in *IstioValidationsService) fetchAllVirtualServices(istioDetails *kubernetes.IstioDetails, namespace string, errChan chan error, wg *sync.WaitGroup) {
+	defer wg.Done()
+	if len(errChan) == 0 {
+		nss, err := in.businessLayer.Namespace.GetNamespaces()
+		if err != nil {
+			errChan <- err
+			return
+		}
+
+		currentVSs, err := in.fetchVirtualServices(namespace)
+		if err != nil {
+			select {
+			case errChan <- err:
+			default:
+			}
+		} else {
+			for _, ns := range nss {
+				vsList, err := in.fetchVirtualServices(ns.Name)
+				if err != nil {
+					select {
+					case errChan <- err:
+					default:
+					}
+				} else {
+					for _, vs := range currentVSs {
```

Comment:
Shouldn't we use the `vsList`? Because we want to go through all VS in the mesh (or available ns). Plus, I guess for each iteration on the `nss` loop, we'd be checking the exact array of VS (from the current ns).

Reply:
'vsList' is the list of VS from exportedTo 'namespace'.
So here it loops through the VS from current namespace, and finds a VS which has 'exportTo' fields.